indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -252,7 +252,7 @@ static void addSnapshotJarToClassPath(
     final Path hdfsPath = new Path(intermediateClassPath, jarFile.getName());
 
     // existing is used to prevent uploading file multiple times in same run.
-    if (!existing.contains(hdfsPath)) {
+    if (!existing.contains(hdfsPath) || !fs.exists(hdfsPath)) {
```

Comment:
exists check is not done for SNAPSHOT jars intentionally so that they get updated everytime. SNAPSHOT jars are to be used for testing only and are designed to get updated on each run. Or else there would be no reason for method 'addSnapshotJarToClassPath' to exist.
i guess your failure is a result of running concurrent jobs with SNAPSHOT jars.